extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -413,3 +420,105 @@ export async function attemptRerun(
+      name: string;
+    }
+  >
+}
+
+/**
+ * Lists the workflow run artifacts for the given workflow run ID.
+ * @returns An object containing the list of artifacts.
+ */
+async function listWorkflowRunArtifacts(
+  credentials: Credentials,
+  owner: string,
+  repo: string,
+  workflowRunId: number
+): Promise<WorkflowRunArtifactsResponse> {
+  const octokit = await credentials.getOctokit();
+  const response = await octokit.rest.actions.listWorkflowRunArtifacts({
+    owner,
+    repo,
+    run_id: workflowRunId,
+  });
+
+  return response.data;
+}
+
+/**
+ * @returns The artifact ID corresponding to the given artifact name.
+ */
+function getArtifactIDfromName(artifactName: string, workflowRunArtifactsResponse: WorkflowRunArtifactsResponse): number | undefined {
+  for (const artifact of workflowRunArtifactsResponse.artifacts) {
+    if (artifact.name === artifactName) {
+      return artifact.id;
+    }
+  }
+  void logger.log(`Could not find artifact with name ${artifactName}.`);
+  return;
+}
+
+/**
+ * Downloads an artifact from a workflow run.
+ * @returns The path to enclosing directory of the unzipped artifact.
+ */
+async function downloadArtifact(
+  credentials: Credentials,
+  owner: string,
+  repo: string,
+  artifactId: number
+): Promise<string> {
+  const octokit = await credentials.getOctokit();
+  const response = await octokit.rest.actions.downloadArtifact({
+    owner,
+    repo,
+    artifact_id: artifactId,
+    archive_format: 'zip',
+  });
+  const artifactPath = path.join(tmpDir.name, `${artifactId}`);
+  void logger.log(`Downloading artifact to ${artifactPath}.zip`);
+  await fs.writeFile(`${artifactPath}.zip`, Buffer.from(response.data as ArrayBuffer));
+
+  void logger.log(`Extracting artifact to ${artifactPath}`);
+  await (await unzipper.Open.file(`${artifactPath}.zip`)).extract({ path: artifactPath });
+  return artifactPath;
+}
+
+interface ResultIndexItem {
+  nwo: string;
+  id: string;
+  results_count: number;
+  bqrs_file_size: number;
+  sarif_file_size?: number;
+}
+
+/**
+ * @param workflowRunId The ID of the workflow run to get the result index for.
+ * @returns An object containing the result index.
+ */
+export async function getResultIndex(
+  credentials: Credentials,
+  owner: string,
+  repo: string,
+  workflowRunId: number
+): Promise<ResultIndexItem[]> {
+  const artifactList = await listWorkflowRunArtifacts(credentials, owner, repo, workflowRunId);
+  const artifactId = getArtifactIDfromName('result-index', artifactList);
+  if (!artifactId) {
+    void showAndLogWarningMessage(
+      `Could not find a result index for the [specified workflow](https://github.com/${owner}/${repo}/actions/runs/${workflowRunId}).
+      Please check whether the workflow run has successfully completed.`
+    );
+    return [];
+  }
+  const artifactPath = await downloadArtifact(credentials, owner, repo, artifactId);
+  const resultIndex = await fs.readFile(path.join(artifactPath, 'index.json'), 'utf8');
```

Comment:
Do we need to check for the existence of index.json first?

Reply:
Done!